meteor/server/api/showStyles.ts

```diff
@@ -75,50 +75,94 @@ export async function insertShowStyleBaseInner(organizationId: OrganizationId |
 		_rundownVersionHash: '',
 	}
 	ShowStyleBases.insert(showStyleBase)
-	await insertShowStyleVariantInner(showStyleBase, 'Default')
+
+	const showStyleVariant: ShowStyleVariant = {
+		_id: getRandomId(),
+		name: 'Default',
+		blueprintConfigWithOverrides: wrapDefaultObject({}),
+		_rundownVersionHash: '',
+		_rank: 500,
+		showStyleBaseId: showStyleBase._id,
+	}
+	await insertShowStyleVariant(showStyleVariant)
 	return showStyleBase._id
 }
-export async function insertShowStyleVariant(
-	context: MethodContext | Credentials,
-	showStyleBaseId: ShowStyleBaseId,
-	name?: string
-): Promise<ShowStyleVariantId> {
+async function assertShowStyleBaseAccess(context: MethodContext | Credentials, showStyleBaseId: ShowStyleBaseId) {
 	check(showStyleBaseId, String)
 
 	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
 	const showStyleBase = access.showStyleBase
 	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
-
-	return insertShowStyleVariantInner(showStyleBase, name)
 }
-export async function insertShowStyleVariantInner(
-	showStyleBase: ShowStyleBaseLight,
+
+export async function createDefaultShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleBaseId: ShowStyleBaseId,
 	name?: string
 ): Promise<ShowStyleVariantId> {
-	return ShowStyleVariants.insertAsync({
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
+
+	const highestRank = ShowStyleVariants.find(
+		{},
+		{
+			sort: {
+				_rank: -1,
+			},
+			limit: 1,
+		}
+	).fetch()[0]?._rank
+	const rank = highestRank !== undefined ? highestRank + 1 : 0
+
+	const showStyleVariant: ShowStyleVariant = {
 		_id: getRandomId(),
-		showStyleBaseId: showStyleBase._id,
 		name: name || 'Variant',
 		blueprintConfigWithOverrides: wrapDefaultObject({}),
 		_rundownVersionHash: '',
-	})
+		_rank: rank,
+		showStyleBaseId: showStyleBaseId,
+	}
+
+	return insertShowStyleVariant(showStyleVariant)
+}
+
+export async function importShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleVariant: ShowStyleVariant
+): Promise<ShowStyleVariantId> {
+	await assertShowStyleBaseAccess(context, showStyleVariant.showStyleBaseId)
+
+	return insertShowStyleVariant(showStyleVariant)
 }
+
+export async function importShowStyleVariantAsNew(
+	context: MethodContext | Credentials,
+	showStyleVariant: ShowStyleVariant
```

Comment:
Minor note: Since the `_id` is overwritten anyway, `Omit<ShowStyleVariant, '_id'>` is a slightly better argument.